src/arrayMethodJoin.js

```diff
@@ -4,9 +4,34 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let result = '';
+    let sep = separator;
+
+    if (separator === null) {
+      sep = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== this[this.length - 1]) {
+        if (this[i] !== undefined && this[i] !== null) {
+          result += this[i] + sep;
+          continue;
+        }
```

Comment:
nested if's are evil 👿 
Any ideas on how we can avoid them here?

Reply:
I was a bit frustrated and wanted it to work 😄 Now, with a clear mind I think I got it.